src/components/Metrics/TrafficDetails.tsx

```diff
@@ -0,0 +1,221 @@
+import { Col, Row } from 'patternfly-react';
+import * as React from 'react';
+import { NodeType } from '../../types/Graph';
+import DetailedTrafficList, { TrafficItem, TrafficNode } from '../Details/DetailedTrafficList';
+import { DurationInSeconds } from '../../types/Common';
+import { getName } from '../../utils/RateIntervals';
+import { MetricsObjectTypes } from '../../types/Metrics';
+
+type WorkloadProps = {
+  itemType: MetricsObjectTypes.WORKLOAD;
+  namespace: string;
+  workloadName: string;
+};
+
+type AppProps = {
+  itemType: MetricsObjectTypes.APP;
+  namespace: string;
+  appName: string;
+};
+
+type TrafficDetailsProps = {
+  rateInterval: DurationInSeconds;
+  trafficData: any;
```

Comment:
Can we have proper typing here, or is it that much of a pain?

Reply:
@rubenvp8510 is working in the typing in #971. I hope his PR can go soon, because I also want to take advantage of those typings. So, let's put some pressure on him :)

(Although, I see he is not adding typing on `API.getNodeGraphElements` in his PR)